src/mon/Monitor.cc

```diff
@@ -2271,6 +2271,110 @@ void Monitor::do_health_to_clog(bool force)
   health_status_cache.summary = summary;
 }
 
+namespace {
+  enum class scrubbed_or_deepscrubbed_t { SCRUBBED, DEEPSCRUBBED };
+
+  void print_unscrubbed_detailed(const std::pair<const pg_t,pg_stat_t> pg_entry,
+				 stringstream &ss,
+				 Formatter *f,
+				 scrubbed_or_deepscrubbed_t how_scrubbed) {
+
+    const auto& pg_stat(pg_entry.second);
+
+    if (f != nullptr) {
+      f->open_object_section("unscrubbed_pg");
+      f->dump_stream("pgid") << pg_entry.first;
+      if (how_scrubbed == scrubbed_or_deepscrubbed_t::SCRUBBED) {
+	f->dump_stream("last_scrub_stamp") << pg_stat.last_scrub_stamp;
+      } else if (how_scrubbed == scrubbed_or_deepscrubbed_t::DEEPSCRUBBED) {
+	f->dump_stream("last_deep_scrub_stamp")
+	  << pg_stat.last_deep_scrub_stamp;
+      }
+      f->close_section();
+    } else {
+      ss << "pg " << pg_entry.first << " is not ";
+      if (how_scrubbed == scrubbed_or_deepscrubbed_t::SCRUBBED) {
+	ss << "scrubbed, last_scrub_stamp "
+	   << pg_stat.last_scrub_stamp;
+      } else if (how_scrubbed == scrubbed_or_deepscrubbed_t::DEEPSCRUBBED) {
+	ss << "deep-scrubbed, last_deep_scrub_stamp "
+	   << pg_stat.last_deep_scrub_stamp;
+      }
+      ss << std::endl;
+    }
+  }
+
+
+  enum class scrubbed_pgs_info_t { COUNT, DETAILED };
+  using pg_stat_map_t = const ceph::unordered_map<pg_t,pg_stat_t>;
+
+  void print_unscrubbed_pgs(pg_stat_map_t& pg_stats,
+			    stringstream &ss,
```

Comment:
this ss shouldn't be needed.  simply generating a health warning from PGMonitor::get_health() will make it show up in ceph -s